### PR TITLE
[stdlib] Remove FileCheck in `test_python_to_mojo.mojo`

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
Related to #2024

### CODE REVIEW

File: stdlib/test/python/test_python_to_mojo.mojo [hunk 1]
```diff
@@ -27,41 +28,38 @@ fn test_string_to_python_to_mojo(inout python: Python) raises:
...
     var end: PythonObject = 4
-    # CHECK: 0
-    # CHECK-NEXT: 1
-    # CHECK-NEXT: 2
-    # CHECK-NEXT: 3
+    expected = List[Int](0, 1, 2, 3)
     for i in range(start, end):
-        print(i)
+        assert_equal(i, expected.pop(0))
+    assert_false(expected)
```
Comment (JoeLoser) [inline stdlib/test/python/test_python_to_mojo.mojo:45]: **Question** Did you mean to have this assertion?  Ditto below for the `assert_false(expected)`.

**Review by JoeLoser — APPROVED** [review]
Looks good, but see my comment wrt use of `%mojo-no-debug` please.

File: stdlib/test/python/test_python_to_mojo.mojo (hunk 1 above)
Comment (gabrieldemarmiesse) [reply stdlib/test/python/test_python_to_mojo.mojo:45]: We check that the list is empty with this. That allows us to make sure that there were only 4 iterations of the loop. Not more, not less

**Comment by gabrieldemarmiesse** [comment]
@JoeLoser done :)

**Comment by JoeLoser** [comment]
> @JoeLoser done :)

Merged, thank you for continuing to push on all these! 😄

**Pull request merged**